stan/math/prim/arr/fun/matrix_gpu.hpp

```diff
@@ -25,169 +20,195 @@ enum copy_transposed_triangular {LOWER_TO_UPPER_TRIANGULAR = 0,
 namespace stan {
   namespace math {
 
+    enum triangularity {LOWER = 0, UPPER = 1, NONE = 2 };
+    enum copy_transposed_triangular {LOWER_TO_UPPER_TRIANGULAR = 0,
+      UPPER_TO_LOWER_TRIANGULAR = 1};
+
+    /**
+     * This class represents a matrix on the GPU. 
+     * 
+     * The matrix data is stored in the oclBuffer.     
+     * 
+     */
     class matrix_gpu {
       private:
-
         cl::Buffer oclBuffer;
 
       public:
+        int rows_;
```

Comment:
`rows_` and `cols_` should also be private if you have a `rows()` and `cols()` accessor.  Does anything need to set them?  Can they be constant?

Reply:
The previous version of the transpose function, these values were modified. Since the change of the transpose to copy on output, this is no longer needed, but was not changed to const. Fixed now. Thank you!